### PR TITLE
Feature/books

### PR DESCRIPTION
**Pull request by immortaldogg**
Add CRUD for books + association between books and categories
- Can create/get/update/delete books
- Validate name/description length and category_id existence before creating a book
- Delete a category will lead to deleting all books associated with that category

### CODE REVIEW

File: service/book/mw_validation.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+		return err
+	}
+	if !catExisted {
+		return ErrCategoryNotExisted
+	}
+
+	return mw.Service.Create(ctx, book)
+}
+
+// func (mw validationMiddleware) FindAll(ctx context.Context, nameToFind string) ([]domain.Book, error) {
```
Comment (fuatto) [inline service/book/mw_validation.go]: Either remove not using code or uncomment it

File: service/book/error.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	ErrRecordNotFound        = errRecordNotFound{}
+	ErrCategoryNotExisted    = errCategoryNotExisted{}
+)
+
+type errDescriptionIsRequired struct{}
+
+func (errDescriptionIsRequired) Error() string {
+	return "book description is required"
+}
+
```
Comment (fuatto) [inline service/book/error.go:22]: Should have implement StatusCode()

**Review by fuatto — CHANGES_REQUESTED** [review]

File: service/book/mw_validation.go (hunk 1 above)
Comment (immortaldogg) [reply service/book/mw_validation.go]: Resolve in newest commit

File: service/book/error.go (hunk 2 above)
Comment (immortaldogg) [reply service/book/error.go:22]: Resolve in newest commit

**Pull request merged**